test/utils.test.ts

```diff
@@ -650,4 +583,63 @@ describe('utils', function () {
       expect(result).to.be.true;
     });
   });
+
+  describe('getRelayClientGenerator', function () {
+    let relayGenerator: Generator<
+      {
+        relayClient: RelayClient;
+        status: string;
+      },
+      never,
+      unknown
+    >;
+
+    beforeEach(function () {
+      relayGenerator = getRelayClientGenerator();
+    });
+
+    it('Should iterate the list of preferred relays and return a preconfigured RelayClient instance each time', function () {
+      for (let i = 0; i < preferredRelays.length; i++) {
+        const nextRelayClient = relayGenerator.next().value.relayClient;
+
+        expect(nextRelayClient.getRelayServerUrl()).equal(preferredRelays[i]);
+      }
+    });
+
+    it('Should return status OK if there are more relays in the list', function () {
+      for (let i = 0; i < preferredRelays.length - 1; i++) {
+        const next = relayGenerator.next().value;
+
+        expect(next.status).equal('OK');
+      }
+    });
+
+    it('Should return a message indicating that the list of relays has reached the end', function () {
+      let next: IteratorResult<
+        {
+          relayClient: RelayClient;
+          status: string;
+        },
+        never
+      >;
+
+      for (let i = 0; i < preferredRelays.length; i++) {
+        next = relayGenerator.next();
+      }
+
+      expect(next!.value.status).equal('End of list of preferred relays');
+    });
+
+    it('Should restar from the first element, once the list of relays has reached the end, ', function () {
+      for (let i = 0; i < preferredRelays.length; i++) {
+        relayGenerator.next();
+      }
+
+      const next = relayGenerator.next();
+
+      expect(next.value.relayClient.getRelayServerUrl()).equal(
+        preferredRelays[0]
+      );
+    });
```

Comment:
Should we add a test where we use the generator in a for, without explicitly calling `next`?

Reply:
@AndresQuijano I think this is still valid

Reply:
Now I included a test without calling next().